src/runtime/audience-action-local-flow.ts

```diff
@@ -304,35 +326,61 @@ export class AudienceActionLocalFlow implements AudienceActionFlow {
     }
   }
 
-  private async initRewardedAdWall_() {
-    this.eventManager_.logSwgEvent(AnalyticsEvent.IMPRESSION_REWARDED_AD);
+  private async googletagReady_(): Promise<boolean> {
+    for (let i = 0; i < this.detectGptRetries_; i++) {
+      if (this.deps_.win().googletag?.apiReady !== undefined) {
+        return this.deps_.win().googletag.apiReady;
+      }
+      await new Promise((r) => setTimeout(r, this.detectGptRetriesMs_));
```

Comment:
User perceivable delay up to 5s seems pretty long. Have you considered delaying initiation of the flow by respecting the client config, `clientDisplayTrigger`? You just need to register the prompt type in the list: https://github.com/subscriptions-project/swg-js/blob/7618e056fab16fb0b465d090a157ca2fe2fe8f97/src/runtime/auto-prompt-manager.ts#L888

Reply:
Rewarded ads should already be added to isActionPromptWithDelay_. There are a number of cases I considered regarding delay:

Contribution:
Normal client - gpt.js should already be loaded because of the `clientDisplayTrigger` delay
Has fake gpt.js - should instantly be detected(no 5 second delay)
Has gpt.js completely blocked - will have delay, but it should be a small portion of readers

Subscription:
There is potentially delay for all clients, but I also think we want to lock the page immediately.

Reply:
Ah, good to know rewarded ads is already added. Just out of curiosity, I believe the delay is 10s for the alpha publisher. Wasn't it sufficient to wait for gpt.js to fully load? Do you think waiting for additional 5s would resolve the issue a lot?

Reply:
For prompts that already have a delay, probably not. I can skip the re-try if we detect that this is a contribution prompt.

Reply:
Would it be more accurate to check if it's closable? (covers all premon/contribution/subscription openaccess)